core/ArchiveProcessor/Rules.php

```diff
@@ -356,4 +364,41 @@ public static function shouldProcessSegmentsWhenReArchivingReports()
     {
         return Config::getInstance()->General['rearchive_reports_in_past_exclude_segments'] != 1;
     }
+
+    public static function isSegmentPluginArchivingDisabled($pluginName, $siteId = null)
+    {
+
+        $config = Config::getInstance();
+
+        //general setting
+        $pluginArchivingSetting = $config->General['disable_archiving_segment_for_plugins'];
+
+        if (!empty($pluginArchivingSetting)) {
+            if (is_string($pluginArchivingSetting)) {
+                $pluginArchivingSetting = explode(",", "$pluginArchivingSetting");
+            }
+            if (in_array($pluginName, $pluginArchivingSetting)) {
+                return true;
+            }
+        }
+
+        if (!$siteId) {
+            return false;
+        }
```

Comment:
Shouldn't this part now be obsolete? The `GeneralConfig` below should now return either the site specific or the global config